projects/go-lib/src/lib/components/go-accordion/go-accordion.component.ts

```diff
@@ -41,24 +42,27 @@ export class GoAccordionComponent implements OnInit, AfterContentInit, OnDestroy
   }
 
   ngAfterContentInit(): void {
-    this.panels.toArray().forEach((panel: GoAccordionPanelComponent, index: number) => {
-      this.updatePanelState(panel, index);
-      this.subscribePanel(panel);
-      panel.detectChanges();
+    this.updateChildPanels();
+
+    this.panels.changes.pipe(takeUntil(this.destroy$)).subscribe(() => {
```

Comment:
Should we look at adding these to a local `subscriptions` variable and unsubscribe from them on destroy?

Reply:
The `takeUntil(this.destroy$)` completes the subscription when the component is destroyed thus cleaning up any outstanding memory leaks.